Makefile

```diff
@@ -19,40 +20,44 @@ ifndef GIT_COMMIT
   GIT_COMMIT := git-$(shell git rev-parse --short HEAD)
 endif
 
-# Source Files
-ALL_FILES = $(shell find ${MKFILE_DIR}${cmd,pkg} -type f -name "*.go")
-CLIENT_FILES = $(shell find ${MKFILE_DIR}${cmd/client,pkg} -type f -name "*.go")
-SERVER_FILES = $(shell find ${MKFILE_DIR}${cmd/server,pkg} -type f -name "*.go")
+# Build Flags
+GO_LD_FLAGS= "-s -w -X github.com/megaease/easegress/pkg/version.RELEASE=${RELEASE} -X github.com/megaease/easegress/pkg/version.COMMIT=${GIT_COMMIT} -X github.com/megaease/easegress/pkg/version.REPO=${GIT_REPO_INFO}"
 
 # Targets
-TARGET_SERVER=${MKFILE_DIR}bin/easegress-server
-TARGET_CLIENT=${MKFILE_DIR}bin/egctl
-TARGET=${TARGET_SERVER} ${TARGET_CLIENT}
+TARGET_SERVER=${RELEASE_DIR}/easegress-server
+TARGET_CLIENT=${RELEASE_DIR}/egctl
 
 # Rules
-build: ${TARGET}
+build: build_client build_server
 
-build_client: ${TARGET_CLIENT}
+build_client:
```

Comment:
Actually the `${TARGET_CLIENT}` will save the meaningless compilation if the source files didn't change. Why eliminate them?

Reply:
Golang itself achieves this.